src/core/grid_based_algorithms/lbgpu.hpp

```diff
@@ -180,6 +179,7 @@ typedef struct {
   unsigned int *seed;
   /** flag indicating whether this site belongs to a boundary */
   unsigned int *boundary;
+  void *philox_state;
```

Comment:
Please find a type safe solution for this. I think you can just move the struct out of the header. Also I'd prefer a more ideomatic name

Reply:
It's going away anyway with the switch to the stateless variant discussed above.